# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [1]:
library(tidyverse) 

-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
<U+221A> ggplot2 2.2.1     <U+221A> purrr   0.2.4
<U+221A> tibble  1.4.2     <U+221A> dplyr   0.7.4
<U+221A> tidyr   0.8.0     <U+221A> stringr 1.3.0
<U+221A> readr   1.1.1     <U+221A> forcats 0.3.0
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [2]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1         V2         V3        V4         V5         V6        V7       
1 0.57158842 0.06556845 0.9901817 0.83940454 0.06338868 0.1946178 0.1572910
2 0.04362633 0.50534262 0.1925122 0.05090723 0.30996555 0.7389461 0.7861143
  V8        V9        V10       V11       V12       V13       V14      
1 0.5234441 0.5975984 0.8465392 0.7061424 0.8182752 0.7399681 0.9123617
2 0.8614819 0.2715878 0.3097504 0.4917659 0.9057509 0.3847882 0.8488862
  V15       V16       V17       V18        V19       V20      
1 0.8291889 0.9257139 0.6814238 0.18124206 0.4828440 0.8266183
2 0.4177742 0.2807136 0.2916938 0.08217766 0.1833631 0.7742783

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [3]:
df2 <- df1  %>%
  gather(Key, value, V1:V20)

head(df2)

Key value     
1 V1  0.57158842
2 V1  0.04362633
3 V2  0.06556845
4 V2  0.50534262
5 V3  0.99018175
6 V3  0.19251220

## Assignment 2

We will generate some data:

In [4]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [5]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [6]:
df5 <- full_join(df3, df4, by = "id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook